### PR TITLE
Start moving some simpler things to new tracker package [API-8].

### PR DESCRIPTION
**Pull request by i509VCB**
Moves Bone meal handling and DragonFightManager tracker hooks to the tracker mixin package.

Also converted the large overwrite in the old DragonFightManagerMixin and replaced it with a redirect (gabizou you'll want to make sure I didn't royally screw up here lol) since the reference to process the dragonSpawnState only occurs once.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/tracker/item/BoneMealItemMixin_Tracker.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
         method = "applyBonemeal",
         at = @At(
             value = "INVOKE",
-            target = "Lnet/minecraft/block/IGrowable;grow(Lnet/minecraft/world/World;Ljava/util/Random;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/state/IBlockState;)V"
+            target = "Lnet/minecraft/block/IGrowable;grow(Lnet/minecraft/world/World;Ljava/util/Random;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/BlockState;)V"
         ),
         require = 0, // Will be removed once the above github issue is resolved with a proper solution
         // Even though we're in a group, expecting this to succeed in forge environments will not work since there is a different mixin
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/tracker/item/BoneMealItemMixin_Tracker.java:80]: Update the redirect method name to like `tracker$wrapGrowWithPhaseEntry`

File: src/main/java/org/spongepowered/common/mixin/tracker/world/end/DragonFightManagerMixin_Tracker.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @param dragonSpawnState The dragon spawn state.
+     * @param worldIn The world this respawnState is occuring in.
+     * @param manager The current DragonFightManager.
+     * @param crystals List of all currently present end crystals.
+     * @param respawnStateTicks The amount of this this respawn state has been running for.
+     * @param exitPortalLocation The position of the exit portal.
+     */
+    @Redirect(method = "tick()V", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/end/DragonSpawnState;process(Lnet/minecraft/world/server/ServerWorld;Lnet/minecraft/world/end/DragonFightManager;Ljava/util/List;ILnet/minecraft/util/math/BlockPos;)V"))
+    private void trackDragonRespawn(DragonSpawnState dragonSpawnState, ServerWorld worldIn, DragonFightManager manager, List<EnderCrystalEntity> crystals, int respawnStateTicks, BlockPos exitPortalLocation) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/tracker/world/end/DragonFightManagerMixin_Tracker.java]: The method name should follow the wiki conventions still: `tracker$wrapSpawnStateWithPhaseEntry` or think of a better name 😆

**Review by gabizou — CHANGES_REQUESTED** [review]
The new redirects/injects should have improved method names, otherwise, good stuff!

File: src/main/java/org/spongepowered/common/mixin/tracker/world/end/DragonFightManagerMixin_Tracker.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package org.spongepowered.common.mixin.tracker.world.end;
```
Comment (ImMorpheus) [inline src/main/java/org/spongepowered/common/mixin/tracker/world/end/DragonFightManagerMixin_Tracker.java:25]: Missing license

**Review by ImMorpheus — CHANGES_REQUESTED** [review]

**Review by gabizou — APPROVED** [review]
I'll get to merge this in tonight.

**Review by ImMorpheus — APPROVED** [review]

**Pull request merged**